substrate/frame/balances/src/impl_currency.rs

```diff
@@ -487,6 +492,24 @@ where
 		)
 		.unwrap_or_else(|_| SignedImbalance::Positive(Self::PositiveImbalance::zero()))
 	}
+
+	fn transferrable_balance(who: &T::AccountId, preservation: Preservation) -> Self::Balance {
```

Comment:
Would it be possible to factor this one function out into a `LegacyCurrencyTransferrableBalance` trait or something?  
Than you can require that in `pallet_evm` like `type Currency: Currency<Balance> + LegacyCurrencyTransferrableBalance<Balance>;` or so.  
It seems like a better solution than editing the existing trait. Do you think it could work?